yi-frontend-vty/src/Yi/Frontend/Vty.hs

```diff
@@ -103,23 +126,36 @@ data FrontendState = FrontendState
 -- If this is set to its default (None) it will be replaced by the default
 -- vty configuration from standardIOConfig. However, standardIOConfig
 -- runs in the IO monad so we cannot set the real default here.
+#if MIN_VERSION_vty(6,0,0)
+newtype BaseVtyConfig = BaseVtyConfig { _baseVtyConfig' :: Maybe Vty.VtyUserConfig }
+#else
 newtype BaseVtyConfig = BaseVtyConfig { _baseVtyConfig' :: Maybe Vty.Config }
+#endif
     deriving (Typeable, Default)
 
 instance YiConfigVariable BaseVtyConfig
 
 makeLenses ''BaseVtyConfig
 
+#if MIN_VERSION_vty(6,0,0)
+baseVtyConfig :: Lens' Config (Maybe Vty.VtyUserConfig)
+#else
 baseVtyConfig :: Lens' Config (Maybe Vty.Config)
+#endif
 baseVtyConfig = configVariable . baseVtyConfig'
 
 start :: UIBoot
 start config submitEvents submitActions editor = do
     let baseConfig = view baseVtyConfig config
+#if MIN_VERSION_vty(6,0,0)
+    vty <- Vty.mkVty . (<> fromMaybe Vty.defaultConfig baseConfig) =<< Vty.userConfig
```

Comment:
Two things:
1. I think it's safe to use `defaultConfig` instead of `standardIOConfig` in vty 6, but I'm not 100% sure, since the migration section in its changelog didn't even talk about it.
2. I kept `userConfig` here to match the old behavior as closely as possible. If you never want vty to be configured that way, then you could do this instead:
```haskell
    vty <- Vty.mkVty (fromMaybe Vty.defaultConfig baseConfig)
```